pkg/asset/machines/kubevirt/machinesets.go

```diff
@@ -0,0 +1,77 @@
+// Package kubevirt generates Machine objects for kubevirt.
+package kubevirt
+
+import (
+	"fmt"
+
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+
+	"github.com/openshift/installer/pkg/types"
+	"github.com/openshift/installer/pkg/types/kubevirt"
+)
+
+// MachineSets returns a list of machinesets for a machinepool.
+func MachineSets(clusterID string, config *types.InstallConfig, pool *types.MachinePool, osImage, role,
+	userDataSecret string) ([]*machineapi.MachineSet, error) {
+
+	if configPlatform := config.Platform.Name(); configPlatform != kubevirt.Name {
+		return nil, fmt.Errorf("non-kubevirt configuration: %q", configPlatform)
+	}
+	if poolPlatform := pool.Platform.Name(); poolPlatform != kubevirt.Name {
+		return nil, fmt.Errorf("non-kubevirt machine-pool: %q", poolPlatform)
+	}
+	platform := config.Platform.Kubevirt
+
+	total := int64(0)
+	if pool.Replicas != nil {
+		total = *pool.Replicas
+	}
+
+	provider := provider(clusterID, platform, pool, userDataSecret, osImage)
+	name := fmt.Sprintf("%s-%s-%d", clusterID, pool.Name, 0)
```

Comment:
What is the significance of the "0" in the name?

Reply:
Its counter to allowcreation of more of one machineset, its not necessary here as we create one machineset - removed